tiledb/sm/filesystem/posix.h

```diff
@@ -259,6 +264,27 @@ class Posix : public FilesystemBase {
   tuple<Status, optional<std::vector<filesystem::directory_entry>>>
   ls_with_sizes(const URI& uri) const override;
 
+  /**
+   * Lists objects and object information that start with `prefix`, invoking
+   * the FilePredicate on each entry collected and the DirectoryPredicate on
+   * common prefixes for pruning.
+   *
+   * @param parent The parent prefix to list sub-paths.
+   * @param f The FilePredicate to invoke on each object for filtering.
+   * @param d The DirectoryPredicate to invoke on each common prefix for
+   *    pruning. This is currently unused, but is kept here for future support.
+   * @param recursive Whether to recursively list subdirectories.
+   *
+   * Note: the return type LsObjects does not match the other "ls" methods so as
+   * to match the S3 equivalent API.
+   */
+  template <FilePredicate F, DirectoryPredicate D>
+  tuple<Status, optional<LsObjects>> ls_filtered(
+      const URI& parent,
+      F f,
+      D d = accept_all_dirs,
+      bool recursive = false) const;
```

Comment:
Stylistic choice to keep the declarations and definitions separate.  But yes it could be defined here.  I don't think it should be since non of the other non-trivial function is

Reply:
I'm going to address this by moving the impelementation to `ls_scanner.h` where it can be re-used for Win32 more easily.